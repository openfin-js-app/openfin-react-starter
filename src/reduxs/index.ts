import {combineReducers} from 'redux';
import {IApplicationState} from './application/types';
import {IClientState} from './client/types';
import {IConfigState} from './config/types';

import application from './application/reducer';
import { defaultState as applicationDefaultState } from './application/reducer';
import client from './client/reducer';
import { defaultState as clientDefaultState } from './client/reducer';
import config from './config/reducer';
import { defaultState as configDefaultState } from './config/reducer';

export * from './application/actions';
export * from './application/types';
export * from './client/actions';
export * from './client/types';
export * from './config/actions';
export * from './config/context';
export * from './config/types';

export interface IRootState {
    application:IApplicationState,
    client:IClientState,
    config:IConfigState,
}

export const rootDefaultState = {
    application:applicationDefaultState,
    client:clientDefaultState,
    config:configDefaultState,
}

export default (parentWindowState?:IRootState)=>{
    if (parentWindowState){
        return combineReducers({
            application:application(parentWindowState.application),
            client:client(parentWindowState.client),
            config:config(parentWindowState.config),
        });

    }else{
        return combineReducers({
            application:application(),
            client:client(),
            config:config(),
        });
    }
}
