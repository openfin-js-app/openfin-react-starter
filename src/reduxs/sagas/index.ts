import { all, takeEvery, select } from 'redux-saga/effects';

import clientSaga from './client';

export function* handleLogAllActions(action) {
    const state = yield select();
    console.log(`${action.type}`,action,'state after',state);
}

export function* watchAndLogSaga(){
    yield takeEvery('*',handleLogAllActions)
}

export default function* rootSaga(){

    const sagas = [ clientSaga() ];

    if(process.env.REACT_APP_LOG_ACTION === 'true'){
        sagas.unshift(watchAndLogSaga());
    }

    yield all(sagas);

}