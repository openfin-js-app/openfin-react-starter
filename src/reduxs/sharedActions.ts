import {
    APPLICATION_SET_LOADING_MSG,
    APPLICATION_LAUNCH_NEW_WINDOW,
} from './application/actions';

import {
    CLIENT_SET_VALUE,
} from './client/actions';

const sharedActions:string[]=[
    APPLICATION_SET_LOADING_MSG,
    APPLICATION_LAUNCH_NEW_WINDOW,
    CLIENT_SET_VALUE,
];

export default sharedActions;