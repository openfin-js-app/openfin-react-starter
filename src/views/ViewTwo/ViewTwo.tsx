import * as React from 'react';
import { useContext } from 'react';
import { ApplicationContext } from 'react-openfin';
import { connect } from 'react-redux';

import {
    IRootState,
    clientSetValue,
} from '../../reduxs';

import {
    ClientCounter
} from '../../components';

interface IProps {
    clientCount:number,
    actions:{
        handleUpdateClientCount:(count:number)=>void
    }

}

const ViewTwo:React.FunctionComponent<IProps> = (
    {
        clientCount,
        actions:{
            handleUpdateClientCount
        }
    }
)=>{

    const {
        state:{
            winTop, winLeft, winWidth, winHeight,
        }
    } = useContext(ApplicationContext);

    return(
        <React.Fragment>
            <div><span>ViewTwo works</span></div>
            <div><span>X:&lt;{winLeft}&gt;|Y:&lt;{winTop}&gt;|W&lt;{winWidth}&gt;H&lt;{winHeight}&gt;</span></div>
            <ClientCounter
                count={clientCount}
                onChange={handleUpdateClientCount}
            />
        </React.Fragment>
    );
}

export default connect(
    (state:IRootState)=>({
        clientCount : state.client.count,
    }),
    dispatch => ({
        actions:{
            handleUpdateClientCount:(count:number)=>{
                dispatch(clientSetValue({count}))
            }
        }
    })
)(ViewTwo);