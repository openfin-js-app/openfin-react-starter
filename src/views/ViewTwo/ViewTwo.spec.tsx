import * as React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import Button from '@material-ui/core/Button';
import configurestore from 'redux-mock-store';


import ViewTwo from './ViewTwo';

import {rootDefaultState} from '../../reduxs';

const mockStore = configurestore();

const store = mockStore(rootDefaultState);

describe('ViewTwo',()=>{

    let shallow;
    let mount;

    beforeAll(() => {
        mount = createMount();
    });

    afterAll(() => {
        mount.cleanUp();
    });


    beforeEach(() => {
        shallow = createShallow();
    });


    it ('render correctly',()=>{

        const wrapper = mount(<ViewTwo store={store}/>);
        expect(wrapper.find(Button)).toHaveLength(2);

        wrapper.find(Button).forEach((button)=>{
            const props = button.props();
            if ( typeof props.onClick === 'function'){
                props.onClick();
            }
        });
        expect(store.getActions()).toMatchSnapshot();

    })
});