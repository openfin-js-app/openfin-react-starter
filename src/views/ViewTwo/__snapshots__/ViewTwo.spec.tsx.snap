// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewOne render correctly 1`] = `
Object {
  "children": "ViewTwo works",
}
`;
